example/feature_demo/marshal_test.go

```diff
@@ -0,0 +1,64 @@
+package example
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/infobloxopen/protoc-gen-gorm/types"
+)
+
+func TestUnmarshalTypes(t *testing.T) {
+	ts, _ := ptypes.TimestampProto(time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC))
+	marshaller := &runtime.JSONPb{OrigName: true, EmitDefaults: true}
+	for in, expected := range map[string]TestTypes{
+		`{}`: {},
+		`{"api_only_string":"important text"}`:            {ApiOnlyString: "important text"},
+		`{"numbers":[1,2,3,4]}`:                           {Numbers: []int32{1, 2, 3, 4}},
+		`{"optional_string":"something real"}`:            {OptionalString: &wrappers.StringValue{Value: "something real"}},
+		`{"becomes_int":"GOOD"}`:                          {BecomesInt: 1},
+		`{"uuid":"123abc-unchecked-uuid"}`:                {Uuid: &types.UUIDValue{Value: "123abc-unchecked-uuid"}},
+		`{"created_at":"2009-11-17T20:34:58.651387237Z"}`: {CreatedAt: ts},
```

Comment:
bummer you had to create this `ts` variable up above. I'm happy with it the way it is, but you could also create a helper function that only returns the single value so you can put it inline here.

I've done this pattern before where things return an error but I want to put a value inline for table-driven tests like this:

```
// GetThing lives in real code, not test code, and does real things that can return real errors
func GetThing()(Thing, error){...}

// MustGetThing lives in test code, and is useful for being able to pass into things inline when you're not expecting to test the error boundary conditions. Useful for downwind tests that consume or depend on a Thing
func MustGetThing(t *testing.T) Thing {
  thing, err := GetThing()
  if err != nil {
    t.Fatal(err)
  }
  return thing
}
```

Reply:
I've seen the `Must` prefix around. I suppose I can create a func for this test to allow that to be inlined.

Reply:
not important -- up to you.
It'd be useful if you were running different tests with multiple timestamps, which you're not doing. Just wanted to point out the pattern :)